### PR TITLE
Moved content negotiation to AdviceTrait test

### LINKED ISSUE #23: Implement content negotiation correctly
We kind of investigate the Accept header, but we should handle non-problem types.

### PR DESCRIPTION
**Pull request by whiskeysierra**
This fixes #23 (even though it never was broken)

### CODE REVIEW

**Comment by whiskeysierra** [comment]
@lukasniemeier-zalando Can you take a look and tell me where I'm going wrong? Why is my test failing?

**Comment by lukasniemeier-zalando** [comment]
Spring throws a `HttpMediaTypeNotAcceptableException` during handling the exception in our `@ExceptionHandler` because `AbstractMessageConverterMethodProcessor` seems to not like to fulfill non-concrete media types.

**Comment by whiskeysierra** [comment]
> AbstractMessageConverterMethodProcessor seems to not like to fulfill non-concrete media types.

https://jira.spring.io/browse/SPR-10493

**Comment by whiskeysierra** [comment]
https://github.com/spring-projects/spring-framework/blob/master/spring-webmvc/src/main/java/org/springframework/web/servlet/mvc/method/annotation/HttpEntityMethodProcessor.java#L216 is not honoring the content type I was setting before going of to do https://github.com/spring-projects/spring-framework/blob/master/spring-webmvc/src/main/java/org/springframework/web/servlet/mvc/method/annotation/AbstractMessageConverterMethodProcessor.java#L163

File: src/test/java/org/zalando/problem/spring/web/advice/AdviceTraitTest.java [hunk 1]
```diff
@@ -102,17 +103,17 @@ public void buildsIfIncludes() {
...
+    public void buildsEmptyIfNotIncludes() throws HttpMediaTypeNotAcceptableException {
         final ResponseEntity<Problem> result = unit.create(Status.RESET_CONTENT, "",
                 request(APPLICATION_ATOM_XML_VALUE));
 
-        assertThat(result, hasFeature("Status", ResponseEntity::getStatusCode, is(RESET_CONTENT)));
-        assertThat(result.getHeaders().getContentType(), is(nullValue()));
-        assertThat(result.getBody(), is(nullValue()));
+       // assertThat(result, hasFeature("Status", ResponseEntity::getStatusCode, is(RESET_CONTENT)));
+       // assertThat(result.getHeaders().getContentType(), is(nullValue()));
+       // assertThat(result.getBody(), is(nullValue()));
```
Comment (lukasniemeier) [inline src/test/java/org/zalando/problem/spring/web/advice/AdviceTraitTest.java]: remove test?

**Comment by lukasniemeier** [comment]
How does this fix #23 (which was never broken :wink:)?

**Comment by whiskeysierra** [comment]
> How does this fix #23 (which was never broken )?

Well... I think we handle Content Negotiation *more correct* now. Is there even a way to allow clients of the library to support other media types?

**Comment by lukasniemeier-zalando** [comment]
Ah now I see the `NOT_ACCEPTABLE` response in case there is no match :+1:

**Pull request merged**